### PR TITLE
7 post api articles article id comments

### PR DESCRIPTION
**Pull request by AlexandraB123**

### CODE REVIEW

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Please are you able to remove the .only from describe block and resubmit the PR. We have to make sure **all** tests have passed before reviewing. 👍

File: __tests__/api.test.js [hunk 1]
```diff
@@ -106,3 +106,126 @@ describe("/api/articles/:article_id", () => {
...
+    };
+    return request(app)
+      .post("/api/articles/not-an-article/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid id");
+      });
+  });
+  test("POST:400. Sends an appropriate error message when given an invalid username", () => {
```
Comment (mezzDavies) [inline __tests__/api.test.js]: This is a very valid test but error should be a 404 as the username is a valid form but doesn't exist - so "not found".

File: models/articles.models.js [hunk 2]
```diff
@@ -26,3 +27,29 @@ exports.fetchArticleById = (article_id) => {
...
+
+exports.addArticleComment = (article_id, body) => {
+  if (!body.username || !body.body)
+    return Promise.reject({ status: 400, msg: "Bad request" });
+  return this.fetchArticleById(article_id)
+    .then(() => checkUserExists(body.username))
+    .then(() => {
+      const queryString = `
+        INSERT INTO comments (article_id, author, body, votes)
+        VALUES ($1, $2, $3, 0)
```
Comment (mezzDavies) [inline models/articles.models.js]: Votes will default to 0 when a comment is inserted (👀 have look at the seed function to confirm this) so no need to include in the values here.

File: __tests__/api.test.js [hunk 3]
```diff
@@ -106,3 +106,126 @@ describe("/api/articles/:article_id", () => {
...
+      });
+  });
+  test("GET:400 sends an appropriate and error message when given an invalid id", () => {
+    return request(app)
+      .get("/api/articles/not-an-article/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Invalid id");
+      });
+  });
```
Comment (mezzDavies) [inline __tests__/api.test.js:163]: We need to test that the comments are sorted by date with the most recent first - the query will need to be amended slightly at to achieve this.

**Review by mezzDavies — CHANGES_REQUESTED** [review]
Good stuff. Clean and very readable code. 😄  Just a few comments that need to looked at though please then resubmit once you've managed to sort 👍

File: models/articles.models.js (hunk 2 above)
Comment (AlexandraB123) [reply models/articles.models.js]: Fixed (votes and 0 value removed)

File: __tests__/api.test.js (hunk 3 above)
Comment (AlexandraB123) [reply __tests__/api.test.js:163]: Is this referring to code already approved on a previous ticket?

File: __tests__/api.test.js (hunk 1 above)
Comment (AlexandraB123) [reply __tests__/api.test.js]: Amended accordingly

File: __tests__/api.test.js (hunk 3 above)
Comment (AlexandraB123) [reply __tests__/api.test.js:163]: Missed on ticket 6. Test and functionality for ordering by date added

**Review by mezzDavies — APPROVED** [review]
All good 👍

**Pull request merged**